src/commonMain/kotlin/exchange.dydx.abacus/state/manager/AsyncAbacusStateManager.kt

```diff
@@ -414,60 +434,73 @@ class AsyncAbacusStateManager(
         if (stateNotification === null && dataNotification === null) {
             throw Error("Either stateNotification or dataNotification need to be set")
         }
-        loadEnvironments()
+        ConfigFile.values().forEach {
```

Comment:
Since you are refactoring this, consider splitting the config loading/saving logic out and make it a separate class.  Non-blocking

Reply:
created a [ticket](https://linear.app/dydx/issue/TRCL-3428/refactor-loadingsaving-config-files-in-abacus-async-state-manager)